packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -617,23 +625,17 @@ private boolean setPendingMethodCallAndResult(
     return true;
   }
 
-  private void finishWithSuccess(String imagePath) {
+  private void finishWithSuccess(ArrayList<String> imagePath, boolean isMultiImage) {
```

Comment:
This feels a bit hacky to give a list and a boolean indicate you only need to pick the first image. I would keep this split into 2 methods like it was.